src/psyclone/domain/lfric/kernel_interface.py

```diff
@@ -0,0 +1,841 @@
+                    "weights", QrWeightsDataSymbol, dims=[nqp])
+                self._arglist.extend([nfaces, nqp, weights])
+            elif shape == "gh_quadrature_edge":
+                nedges = self._create_symbol("nedges", NumberOfEdgesDataSymbol)
+                nqp = self._create_symbol("nqp", NumberOfQrPointsDataSymbol)
+                weights = self._create_symbol(
+                    "weights", QrWeightsDataSymbol, dims=[nqp])
+                self._arglist.extend([nedges, nqp, weights])
+            else:
+                raise InternalError("Unsupported quadrature shape '{0}' "
+                                    "found in kernel_interface.".format(shape))
+
+    def _create_symbol(self, tag, data_symbol, extra_args=None, dims=None,
+                       interface=None):
+        '''Internal utility to create a symbol. If a symbol is found in the
+        symbol table with the supplied tag then that symbol is
+        returned, otherwise a new symbol of type 'data_symbol' is
+        created and added to the symbol table with the supplied
+        tag. If the symbol requires any arguments then these are
+        supplied via the extra_args and dims arguments. The latter
+        also specifies the dimensions of the symbol if it is an array. By
+        default it is assumed that the access to the symbol will be
+        read only; if the access is different to this then the
+        interface argument must be provided with the appropriate
+        access type.
+
+        As this is an internal utility, we assume that the argument
+        datatypes and content are correct.
+
+        :param str tag: the name to use as a tag in the symbol table \
+            and also as the base name when creating a new symbol table \
+            name.
+        :param data_symbol: the symbol class that we are going to create.
+        :type data_symbol: :py:class:`psyclone.psyir.symbols.DataSymbol`
+        :param extra_args: an optional list of strings specifying the \
+            values of any additional arguments to provide to the \
+            data_symbol class on creation or None if there are \
+            none. Defaults to None.
+        :type extra_args: list of str or NoneType
+        :param dims: an optional list of dimensions used to dimension \
+            the data_symbol class if it is an array, or None if it is \
+            not an array. Defaults to None.
+        :type dims: list of DataSymbol, int or ArrayType.Extent or NoneType
+        :param interface: an optional ArgumentInterface specifying the \
+            intent of the symbol, or None if the default it \
+            suitable. Defaults to read access.
+        :type interface: \
+            :py:class:`psyclone.psyir.symbols.ArgumentInterface` or \
+            NoneType
+
+        :returns: the appropriate symbol and properties as specified \
+            by this methods arguments.
+        :type: :py:class:`psyclone.psyir.symbols.DataSymbol`
```

Comment:
It will return a symbol of type "symbol_type" which should be a subclass of datasymbol (although we don't check). I've changed the return type to be "subclass of ...". However, I do have a general question. Do you think we should put "subclass of" when the method can return objects that are different types but are all a subclass of a parent class? If so, I don't think we consistently do this in PSyclone - in particular when we use `Node` when we don't know the type of PSyIR that is being returned.

Reply:
Hmm. I think it's clearer if we say "subclass of" - we're providing more information which can only help with understanding the code. If @hiker, @TeranIvy and @sergisiso agree then we should update the style guide.

Reply:
OK, I've added some words to the wiki.